Source/WebCore/editing/cocoa/HTMLConverter.mm

```diff
@@ -2613,8 +2613,8 @@ static void updateAttributes(const Node* node, const RenderStyle& style, OptionS
 
     if (linkURL.isEmpty())
         [attributes removeObjectForKey:NSLinkAttributeName];
-    else
-        [attributes setObject:linkURL.createNSURL().get() forKey:NSLinkAttributeName];
+    else if (RetainPtr linkNSURL = linkURL.createNSURL())
```

Comment:
We may want to remove the NSLinkAttributeName key if this is null.  So this could be:
if (RetainPtr linkNSURL = linkURL.createNSURL())
    [attributes setObject:linkNSURL.get() forKey:NSLinkAttributeName];
else
    [attributes setObject:linkURL.createNSURL().get() forKey:NSLinkAttributeName];

Reply:
Hmm I think I see what you mean, you mean to say we should `removeObjectForKey:NSLinkAttributeName` when the URL is invalid too, right?

```cpp
if (RetainPtr linkNSURL = linkURL.createNSURL(); linkURL.isEmpty() || !linkNSURL)
    [attributes removeObjectForKey:NSLinkAttributeName];
else
    [attributes setObject:linkNSURL.get() forKey:NSLinkAttributeName
```